### PR TITLE
Add support to GORM

### PR DESCRIPTION
**Pull request by gandaldf**
I hope this can be useful to the project.
Let me know if I need to fix anything!

### CODE REVIEW

**Comment by ReneWerner87** [comment]
thank you for your work

we will consult internally whether to adopt the pull request

**Comment by gandaldf** [comment]
Thank you! The motivation behind this PR is to handle sessions just as you handle the rest of your data.
Without this feature you have to check the GORM dialect in use and manually select the single store package to handle your sessions; when you have to support 4 RDBMS it could be pretty annoying.

File: gorm/gorm.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+		case <-s.done:
+			return
+		case t := <-ticker.C:
+			s.gc(t)
+		}
+	}
+}
+
+// gc deletes all expired entries
+func (s *Storage) gc(t time.Time) {
```
Comment (efectn) [inline gorm/gorm.go:220]: I think we can use same logic here. https://github.com/gofiber/storage/blob/main/memory/memory.go

New logic is more performant

File: gorm/gorm.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+		// Use passed db
+		db = cfg.Db
+
+		sqldb, err = db.DB()
+		if err != nil {
+			panic(err)
+		}
+	} else {
+		// Create db
+		switch cfg.Dialect {
```
Comment (li-jin-gou) [inline gorm/gorm.go:50]: I'm wondering if we just need to support DB parameters, as adding other parameters isn't necessarily useful.

Comment (gandaldf) [reply gorm/gorm.go:50]: Well, I think that every Storage package should only support a DB parameter initialized outside the package itself.
Many DBs have a lot config options that make it hard to fully support them, other times you just want to share your already initialized data connection for sessions too.
I'm not experienced enough to discuss precedent decisions, so I have just tried to copy the same code already in production but if you need it I can change it...
Maybe I can find some time to align all other Storage packages to support just an external connection pointer.

File: gorm/gorm.go (hunk 1 above)
Comment (gandaldf) [reply gorm/gorm.go:220]: Ok! I can change it... but we should change it in MySQL and all other Storage packages.

**Comment by ReneWerner87** [comment]
the storages should only be hardware close driver and not contain an intermediate layer like an orm model

we made a small voting internally about this and came to the result that we don't merge this pull request for the time being until the community decides otherwise

**Pull request closed without merging**